FilePersistence/src/version_control/merge/ChangeGraph.cpp

```diff
@@ -448,10 +448,20 @@ void ChangeGraph::updateOutgoingLabels(Model::NodeIdType /*parentId*/, IdToLabel
 	Q_ASSERT(false);
 }
 
-void ChangeGraph::updateBaseTreeLabels(Model::NodeIdType /*parentId*/, IdToLabelMap /*labelMap*/,
-													GenericTree* /*baseTree*/)
+void ChangeGraph::updateBaseTreeLabels(Model::NodeIdType parentId, IdToLabelMap labelMap,	GenericTree* baseTree)
 {
-	Q_ASSERT(false);
+	auto parentNode = baseTree->find(parentId);
+
+	// Update baseTree labels according to base Labels in labelMap
+	for (auto node : parentNode->children())
+	{
+		auto labelIt = labelMap.find(node->id());
+		while (labelIt != labelMap.end() && labelIt.key() == node->id()) {
+			if (labelIt.value().branch.testFlag(MergeChange::None))	// Base element
+				break;
+		}
+		node->setLabel(labelIt.value().label);
```

Comment:
I think you are implicitly assuming here that we always hit the `// Base element` and break out of the loop and then update the node with the new label. However, to be on the safe side, we should make sure to only update the label if we indeed hit `// Base element`. In particular, right now we'll update the node, every time we call `lablMap.find` and it returns `end()`.

Just move this update in the `if` above, right before the break.